public/icvar.h

```diff
@@ -11,21 +11,456 @@
+template<> inline void ConVarBaseData_t::ValueToString<Vector>( const Vector& val, char* dst, size_t length )			{ snprintf( dst, length, "%f %f %f", val.x, val.y, val.z ); }
+template<> inline void ConVarBaseData_t::ValueToString<Vector4D>( const Vector4D& val, char* dst, size_t length )		{ snprintf( dst, length, "%f %f %f %f", val.x, val.y, val.z, val.w ); }
+template<> inline void ConVarBaseData_t::ValueToString<QAngle>( const QAngle& val, char* dst, size_t length )			{ snprintf( dst, length, "%f %f %f", val.x, val.y, val.z ); }
+
+template<> inline bool ConVarBaseData_t::ValueFromString<bool>( const char* val )				{ if (strcmp(val, "true") == 0 || strcmp(val, "1") == 0) { return true; } return false; }
+template<> inline uint16_t ConVarBaseData_t::ValueFromString<uint16_t>( const char* val )		{ unsigned int ret; sscanf(val, "%u", &ret); return ret; }
+template<> inline int16_t ConVarBaseData_t::ValueFromString<int16_t>( const char* val )			{ int ret; sscanf(val, "%d", &ret); return ret; }
+template<> inline uint32_t ConVarBaseData_t::ValueFromString<uint32_t>( const char* val )		{ uint32_t ret; sscanf(val, "%u", &ret); return ret; }
+template<> inline int32_t ConVarBaseData_t::ValueFromString<int32_t>( const char* val )			{ int32_t ret; sscanf(val, "%d", &ret); return ret; }
+template<> inline uint64_t ConVarBaseData_t::ValueFromString<uint64_t>( const char* val )		{ uint64_t ret; sscanf(val, "%lu", &ret); return ret; }
+template<> inline int64_t ConVarBaseData_t::ValueFromString<int64_t>( const char* val )			{ int64_t ret; sscanf(val, "%ld", &ret); return ret; }
+template<> inline float ConVarBaseData_t::ValueFromString<float>( const char* val )				{ float ret; sscanf(val, "%f", &ret); return ret; }
+template<> inline double ConVarBaseData_t::ValueFromString<double>( const char* val )			{ double ret; sscanf(val, "%lf", &ret); return ret; }
+template<> inline const char* ConVarBaseData_t::ValueFromString<const char*>( const char* val )	{ return val; }
+template<> inline Color ConVarBaseData_t::ValueFromString<Color>( const char* val )				{ int r, g, b, a; sscanf(val, "%d %d %d %d", &r, &g, &b, &a); return Color(r, g, b, a); }
+template<> inline Vector2D ConVarBaseData_t::ValueFromString<Vector2D>( const char* val )		{ float x, y; sscanf(val, "%f %f", &x, &y); return Vector2D(x, y); }
+template<> inline Vector ConVarBaseData_t::ValueFromString<Vector>( const char* val )			{ float x, y, z; sscanf(val, "%f %f %f", &x, &y, &z); return Vector(x, y, z); }
+template<> inline Vector4D ConVarBaseData_t::ValueFromString<Vector4D>( const char* val )		{ float x, y, z, w; sscanf(val, "%f %f %f %f", &x, &y, &z, &w); return Vector4D(x, y, z, w); }
+template<> inline QAngle ConVarBaseData_t::ValueFromString<QAngle>( const char* val )			{ float x, y, z; sscanf(val, "%f %f %f", &x, &y, &z); return QAngle(x, y, z); }
+
+template<typename T>
+struct ConVarData_t : ConVarBaseData_t

```

Comment:
While it's not too late to change the names, how about change this to a class and remove the ``_t``?